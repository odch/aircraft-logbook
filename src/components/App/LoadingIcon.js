import React from 'react'
import PropTypes from 'prop-types'
import CircularProgress from '@material-ui/core/CircularProgress'
import withStyles from '@material-ui/core/styles/withStyles'

const styles = () => ({
  container: {
    position: 'absolute',
    width: '100%',
    height: '100%'
  },
  icon: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    margin: '-25px 0 0 -25px'
  }
})

const LoadingIcon = props => (
  <div className={props.classes.container}>
    <CircularProgress size={50} className={props.classes.icon} />
  </div>
)

LoadingIcon.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(LoadingIcon)
