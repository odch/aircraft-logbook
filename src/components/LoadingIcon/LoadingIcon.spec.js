import React from 'react'
import renderer from 'react-test-renderer'
import LoadingIcon from './LoadingIcon'

describe('components', () => {
  describe('LoadingIcon', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<LoadingIcon />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
