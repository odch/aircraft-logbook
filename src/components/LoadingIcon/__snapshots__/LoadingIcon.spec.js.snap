// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LoadingIcon renders correctly 1`] = `
<div
  className="LoadingIcon-container-1"
>
  <div
    className="MuiCircularProgress-root-3 MuiCircularProgress-colorPrimary-6 MuiCircularProgress-indeterminate-5 LoadingIcon-icon-2"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-8"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-9 MuiCircularProgress-circleIndeterminate-11"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
